android/BOINC/app/build.gradle

```diff
@@ -1,5 +1,45 @@
+plugins {
+    id 'org.ajoberstar.grgit' version '1.7.1'
+    id 'com.palantir.git-version' version '0.7.1'
+}
+
 apply plugin: 'com.android.application'
 
+import org.ajoberstar.grgit.Grgit
+
+// Use commit date as version code
+def buildVersionCode() {
+    def repo = Grgit.open()
+    def head = repo.head()
+
+    // Sanity check across git plugins
+    assert head.getAbbreviatedId(10) == versionDetails().gitHash : "Internal error: SHA1 mismatch!"
+
+    return head.time
```

Comment:
My statement was just meant as information for someone else stumbling about the `versionCode` limitation. I think it is fine and we don't need a factor. If this is still a problem 19 years from now let the future BOINC developers worry about that. :smiley: